zipline/testing/src/commonMain/kotlin/app/cash/zipline/testing/kotlinSerializers.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2017-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package app.cash.zipline.testing
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.SerialKind
+import kotlinx.serialization.descriptors.StructureKind
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.encodeCollection
+import kotlinx.serialization.modules.SerializersModule
+
+/*
+ *
+ * This file contains built-in serializers from Kotlin Serialization. It's here to work around a gap
+ * in the Kotlin Serialization API - there's no public API like this:
+ *
+ *   fun serializer(
+ *     kClass: KClass<*>,
+ *     typeArgumentsSerializers: List<KSerializer<*>>
+ *   )
+ *
+ * The API gets close! There's an API to get a built-in serializer with a fully-specified KType, and
+ * there's an API to do this for a contextual serializer.
+ *
+ * There's something that's almost perfect but it's private, SerializersModule.builtinSerializer().
+ *
+ * This is used by Zipline when we have a serializer for a type parameter 'T' and need to create a
+ * serializer for a `List<T>`.
+ *
+ * TODO(jwilson): delete this once kotlinx.serialization has a public API for this.
```

Comment:
Is there an issue?

Reply:
I wish I hadn't scrolled this far 😢

Reply:
I found a much better fix. There’s a public API to create a `ListSerializer`

Reply:
(Test-only code either way!)

Reply:
https://github.com/Kotlin/kotlinx.serialization/issues/2025